From: David Steele <dsteele@gmail.com>
Date: Thu, 16 Apr 2015 20:50:35 -0400
Subject: cloudprint.py: Don't flush the C.P. job early

process_job() was calling Cloud Print with a finish_job message
before submitting to CUPS. If the CUPS job failed, the Cloud Print
job would be gone, and the queue would list the job as 'Success'.

By calling Cloud Print only after calling CUPS, failures will be
properly propagated up to the Cloud Print queue.

Conflicts:
	cloudprint/cloudprint.py
---
 cloudprint/cloudprint.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index d72415a..27478d7 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -385,12 +385,12 @@ def process_job(cups_connection, cpp, printer, job):
 
         options = dict((str(k), str(v)) for k, v in options.items())
 
-        cpp.finish_job(job['id'])
-
         cups_connection.printFile(printer.name, tmp.name, job['title'], options)
         os.unlink(tmp.name)
         LOGGER.info('SUCCESS ' + job['title'].encode('unicode-escape'))
 
+        cpp.finish_job(job['id'])
+
     except Exception:
         cpp.fail_job(job['id'])
         LOGGER.exception('ERROR ' + job['title'].encode('unicode-escape'))
